frontend/js/summonerStats.js

```diff
@@ -0,0 +1,218 @@
+    console.log(matches)
+    return matches
+}
+
+const getIcon = async() => {
+    const name = new URL(location.href).searchParams.get("summonerName");
+
+    const icon = await (await fetch(`${BASEURL}/${name}`)).json()
+    console.log(icon)
+    return icon
+}
+
+
+const displayMatches = (matches, info) => {
+    //display infos that don't neet loop
+    getAndDisplayProfile(info);
+    getAndDisplayRank(info);
+    // Loop to access all rows
+    const matchesTabs = matches.map((match) => {
+        console.log(match);
+        // creating Matchtab
+        const matchTab = createMatchTab()
+        const playersLeft = createPlayersLeft()
+        const playersRight = createPlayersRight()
+        const playerScore = createPlayerScore()
+        const matchInfo = createMatchInfo()
+        const playerChampImage = createPlayerChampImage()
+        const playerStats = document.createElement("div")
+        const playerInfo = createPlayerInfo()
+
+        matchTab.append(playerScore)
+            //loop throught players
+        const playerTabs = match.info.participants.map((player) => {
+            const playerTab = createPlayerTab()
+            const champImage = createChampImage(player)
+            const kda = document.createElement("span")
+            playerScore.append(playerChampImage)
+            const playerName = createPlayerName(player)
+            console.log(new URL(location.href).searchParams.get("summonerName"))
+            if (player.summonerName.toLowerCase() == new URL(location.href).searchParams.get("summonerName").toLowerCase()) {
+                getMatchInfo(match, matchInfo, matchTab, player)
+                playerChampImage.append(champImage.cloneNode())
+                playerStats.append(player.kills + "/")
+                playerStats.append(player.deaths + "/")
+                playerStats.append(player.assists)
+                kda.append("KDA: " + ((player.kills + player.assists) / player.deaths).toFixed(2))
+                playerInfo.append(player.totalMinionsKilled + "CS")
+                playerInfo.append(" Wards:" + player.visionScore)
+            }
+            playerScore.append(playerStats)
+            playerScore.append(playerInfo)
+            playerInfo.append(kda)
+            playerTab.append(champImage)
+            playerTab.append(playerName)
+            return playerTab
+
+        })
+
+        const playerTabLeft = playerTabs.slice(0, 5)
+        const playerTabRight = playerTabs.slice(5, 10)
+        playersLeft.append(...playerTabLeft)
+        playersRight.append(...playerTabRight)
+        matchTab.append(playersLeft)
+        matchTab.append(playersRight)
+
+        return matchTab
+        console.log(matchTab)
+
+    })
+    document.getElementById("games").append(...matchesTabs)
+}
+
+const getAndDisplayMatches = async() => {
+    const matches = await getMatches()
+    const iconid = await getIcon()
+    displayMatches(matches.data, iconid.data)
```

Comment:
displayMatches has the only job of displaying matches, remove the icon part and the rank part, rename the function "getAndDisplayMatches" as setupUserPahe (or something similar) and put the logic to display icon and ranks here